connectors/vmware/src/centreon/script/centreon_vmware.pm

```diff
@@ -108,64 +108,131 @@ sub new {
+        # the whole program. The wanted structure is a key-object dictionary instead of an array of objects.
+        my %vsphere_server_dict = map { lc($_->{name}) => $_ } @{$centreon_vmware_config_from_json->{vsphere_server}};
+        # Replace the "raw" structure from the JSON file.
+        $centreon_vmware_config_from_json->{vsphere_server} = \%vsphere_server_dict;
+        # Concatenation of the default parameters with the ones from the config file
+        $self->{centreon_vmware_config} = {%{$self->{centreon_vmware_default_config}}, %$centreon_vmware_config_from_json};
+    } else {
+        $self->{logger}->writeLogError($self->{opt_extra} . " does not seem to be in a supported format.");
+    }
+}
+
+
+# report_config_check: writes a report of what has been loaded from the configuration file
+sub report_config_check {
+    my ($self, %options) = @_;
+
+    my $nb_server_entries = scalar(keys %{$self->{centreon_vmware_config}->{vsphere_server}});
+    my $entry_spelling    = ($nb_server_entries > 1) ? 'entries' : 'entry';
+
+    my $report = "Configuration file " . $self->{opt_extra} . " has been read correctly and has "
+        . $nb_server_entries . " " . $entry_spelling . ".";
+
+    $self->{logger}->writeLogInfo($report);
+    return $report;
+}
+
 sub init {
     my $self = shift;
     $self->SUPER::init();
 
     # redefine to avoid out when we try modules
     $SIG{__DIE__} = undef;
 
-    if (!defined($self->{opt_extra})) {
+    if ( ! defined($self->{opt_extra}) ) {
         $self->{opt_extra} = "/etc/centreon/centreon_vmware.pm";
     }
-    if (-f $self->{opt_extra}) {
-        require $self->{opt_extra};
-    } else {
-        $self->{logger}->writeLogInfo("Can't find extra config file $self->{opt_extra}");
+
+    if ( ! -f $self->{opt_extra} ) {
+        $self->{logger}->writeLogFatal(
+            "Can't find config file '$self->{opt_extra}'. If a migration from "
+            . "/etc/centreon/centreon_vmware.pm to /etc/centreon/centreon_vmware.json is required, you may "
+            . "centreon_vmware_convert_config_file /etc/centreon/centreon_vmware.pm > /etc/centreon/centreon_vmware.json"
+        );
+    }
+
+    if ( ! $self->{opt_extra} =~ /\.(pm|json)$/i) {
+        $self->{logger}->writeLogFatal(
+            "Configuration file " . $self->{opt_extra} . " seems to be neither JSON nor PM file."
+        );
+    }
+
+    $self->read_configuration(filename => $self->{opt_extra});
+
+    if (! defined($self->{opt_vault_config}) or $self->{opt_vault_config} eq '') {
+        $self->{opt_vault_config} = '/var/lib/centreon/vault/vault.json';
+        $self->{logger}->writeLogInfo("No vault config file given. Applying default: " . $self->{opt_vault_config});
     }
 
-    $self->{centreon_vmware_config} = {%{$self->{centreon_vmware_default_config}}, %centreon_vmware_config};
+    # At this point $self->{vault_enabled} can be undef or 0. If 0 we don't enter here
```

Comment:
We could do both, and since you first read the code, the `if` has been changed to be compatible with this idea. Do you think it must be set to 1 now or are you ok with this new version?